geopandas/plotting.py

```diff
@@ -766,12 +770,23 @@ def plot_dataframe(
 
         if classification_kwds is None:
             classification_kwds = {}
-        if "k" not in classification_kwds:
-            classification_kwds["k"] = k
 
-        binning = mapclassify.classify(
-            np.asarray(values[~nan_idx]), scheme, **classification_kwds
-        )
+        if scheme == "greedy":
+            colors = mapclassify.greedy(df, **classification_kwds)
+            num_colors = colors.max() + 1
+            bins = np.arange(num_colors)
+            binning = SimpleNamespace(
```

Comment:
My first attempt was to create make an instance of `mapclassify.classifiers.MapClassifier` to match what `mapclassify.classify` returns but that didn't work as that class is abstract. This namespace approach seemed like a good one to overcome that issue.